essentials-framework/Essentials Core/PepperDashEssentialsBase/Secrets/SecretsManager.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Crestron.SimplSharp;
+using PepperDash.Core;
+
+
+namespace PepperDash.Essentials.Core
+{
+    public static class SecretsManager
+    {
+        public static List<ISecretProvider> Secrets { get; set; }

```

Comment:
Make this a dictionary. It'll be quicker to retrieve via key that way rather than using LINQ.